### PR TITLE
python310Packages.pythran: 0.9.12 -> 0.11.0

### PR DESCRIPTION
**Pull request by K900**
###### Description of changes

Unbreaks latest scipy.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/pythran/0001-hardcode-path-to-libgomp.patch [hunk 1]
```diff
@@ -1,64 +1,91 @@
...
+From a372b91337666023b90b34622248e912d0442497 Mon Sep 17 00:00:00 2001
+From: K900 <me@0upti.me>
+Date: Tue, 28 Jun 2022 10:14:12 +0300
 Subject: [PATCH] hardcode path to libgomp
 
 ---
- omp/__init__.py | 40 ++++------------------------------------
- 1 file changed, 4 insertions(+), 36 deletions(-)
+ omp/__init__.py | 68 ++-----------------------------------------------
+ 1 file changed, 2 insertions(+), 66 deletions(-)
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/pythran/0001-hardcode-path-to-libgomp.patch:9]: We don't need this. We can just export git diff which reduces the update noise.

File: pkgs/development/python-modules/pythran/default.nix [hunk 2]
```diff
@@ -22,13 +22,13 @@ let
...
 
   src = fetchFromGitHub {
     owner = "serge-sans-paille";
     repo = "pythran";
     rev = version;
-    sha256 = "sha256-lQbVq4K/Q8RzlFhE+l3HPCmUGmauXawcKe31kfbUHsI=";
+    sha256 = "sha256-F9gUZOTSuiqvfGoN4yQqwUg9mnCeBntw5eHO7ZnjpzI=";
   };
 
   patches = [
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/pythran/default.nix:34]: Please remove pytest-runner.

**Review by SuperSandro2000 — COMMENTED** [review]
Please fix the commit message.

File: pkgs/development/python-modules/pythran/default.nix [hunk 3]
```diff
@@ -62,14 +55,7 @@ in buildPythonPackage rec {
     "pythran.spec"
   ];
 
-  checkInputs = [
-    pytestCheckHook
-    numpy
-    scipy
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/pythran/default.nix:68]: If we drop this, we should also drop them from the inputs if they are unused.

Comment (K900) [reply pkgs/development/python-modules/pythran/default.nix:68]: I did.

**Pull request merged**